JustLog/Classes/Logger.swift

```diff
@@ -105,47 +119,76 @@ extension Logger: Logging {
-        internalLogger.verbose(logMessage, file, function, line: Int(line))
+        log(.verbose, message, error: error, userInfo: updatedUserInfo, file, function, line)
     }
     
     public func debug(_ message: String, error: NSError?, userInfo: [String : Any]?, _ file: StaticString, _ function: StaticString, _ line: UInt) {
         let file = String(describing: file)
         let function = String(describing: function)
         let updatedUserInfo = [logTypeKey: "debug"].merged(with: userInfo ?? [String : String]())
-        let logMessage = self.logMessage(message, error: error, userInfo: updatedUserInfo, file, function, line)
-        internalLogger.debug(logMessage, file, function, line: Int(line))
+        log(.debug, message, error: error, userInfo: updatedUserInfo, file, function, line)
     }
     
     public func info(_ message: String, error: NSError?, userInfo: [String : Any]?, _ file: StaticString, _ function: StaticString, _ line: UInt) {
         let file = String(describing: file)
         let function = String(describing: function)
         let updatedUserInfo = [logTypeKey: "info"].merged(with: userInfo ?? [String : String]())
-        let logMessage = self.logMessage(message, error: error, userInfo: updatedUserInfo, file, function, line)
-        internalLogger.info(logMessage, file, function, line: Int(line))
+        log(.info, message, error: error, userInfo: updatedUserInfo, file, function, line)
     }
     
     public func warning(_ message: String, error: NSError?, userInfo: [String : Any]?, _ file: StaticString, _ function: StaticString, _ line: UInt) {
         let file = String(describing: file)
         let function = String(describing: function)
         let updatedUserInfo = [logTypeKey: "warning"].merged(with: userInfo ?? [String : String]())
-        let logMessage = self.logMessage(message, error: error, userInfo: updatedUserInfo, file, function, line)
-        internalLogger.warning(logMessage, file, function, line: Int(line))
+        log(.warning, message, error: error, userInfo: updatedUserInfo, file, function, line)
+
     }
     
     public func error(_ message: String, error: NSError?, userInfo: [String : Any]?, _ file: StaticString, _ function: StaticString, _ line: UInt) {
         let file = String(describing: file)
         let function = String(describing: function)
         let updatedUserInfo = [logTypeKey: "error"].merged(with: userInfo ?? [String : Any]())
-        let logMessage = self.logMessage(message, error: error, userInfo: updatedUserInfo, file, function, line)
-        internalLogger.error(logMessage, file, function, line: Int(line))
+        log(.error, message, error: error, userInfo: updatedUserInfo, file, function, line)
+    }
+    
+    internal func log(_ type: LogType, _ message: String, error: NSError?, userInfo: [String : Any]?, _ file: String, _ function: String, _ line: UInt) {
+        switch self.eventLogingPolicy {
```

Comment:
The following lines

```
let logMessage = self.logMessage(message, error: error, userInfo: userInfo, file, function, line)
sendLogMessage(with: type, logMessage: logMessage, file, function, line)
```

are written 3 times. We could remove at least one occurrence like so:

```switch loggingPolicy {
        case .multipleEvents where type == .error:
            if let error  = error {
                for error in error.disassociatedErrorChain() {
                    let logMessage = self.logMessage(message, error: error, userInfo: userInfo, file, function, line)
                    sendLogMessage(with: type, logMessage: logMessage, file, function, line)
                }
            }
        default:
            let logMessage = self.logMessage(message, error: error, userInfo: userInfo, file, function, line)
            sendLogMessage(with: type, logMessage: logMessage, file, function, line)
        }
```